### ISSUE TITLE
CT duplicates are not allowed

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/equality-explorer#159]
```
equality-explorer-basics : fuzz : built
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000
Query: fuzz&memoryLimit=1000
Uncaught Error: duplicates are not allowed
Error: duplicates are not allowed
at kd.push (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:761375)
at Ib.enter (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:843802)
at qc.dispatchToListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:703729)
at qc.dispatchToTargets (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:703904)
at qc.dispatchEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:703494)
at qc.enterEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:703159)
at qc.branchChangeEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:703051)
at qc.moveEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:702633)
at mouseMoveAction.Ur.phetioPlayback (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:690206)
at _r.execute (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599642657625%7D&fuzz&memoryLimit=1000:858:391024)
id: Bayes Chrome
Snapshot from 9/9/2020, 2:33:02 AM

----------------------------------

equality-explorer-basics : fuzz : built
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000
Query: fuzz&memoryLimit=1000
Uncaught Error: duplicates are not allowed
Error: duplicates are not allowed
at kd.push (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:761375)
at Ib.enter (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:843802)
at qc.dispatchToListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:703729)
at qc.dispatchToTargets (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:703904)
at qc.dispatchEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:703494)
at qc.enterEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:703159)
at qc.branchChangeEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:703051)
at qc.moveEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:702633)
at mouseMoveAction.Ur.phetioPlayback (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:690206)
at _r.execute (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/build/phet/equality-explorer-basics_en_phet.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22built%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599646908084%7D&fuzz&memoryLimit=1000:858:391024)
id: Bayes Chrome
Snapshot from 9/9/2020, 2:33:02 AM

----------------------------------

equality-explorer-basics : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/equality-explorer-basics/equality-explorer-basics_en.html?continuousTest=%7B%22test%22%3A%5B%22equality-explorer-basics%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1599640382837%22%2C%22timestamp%22%3A1599641306751%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: duplicates are not allowed
Error: duplicates are not allowed
at ObservableArray.push (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/axon/js/ObservableArray.js:216:13)
at FireListener.enter (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/listeners/PressListener.js:671:23)
at Input.dispatchToListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1852:25)
at Input.dispatchToTargets (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1884:12)
at Input.dispatchEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1810:10)
at Input.enterEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1745:14)
at Input.branchChangeEvents (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1715:12)
at Input.moveEvent (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:1643:10)
at Input.mouseMoveAction.Action.phetioPlayback (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/scenery/js/input/Input.js:316:12)
at Action.execute (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1599640382837/axon/js/Action.js:226:18)
id: Bayes Chrome
Snapshot from 9/9/2020, 2:33:02 AM
```

**Comment by KatieWoe** [comment]
Applies to other Equality Explorer sims as well.

**Comment by pixelzoom** [comment]
CT shows this occurring in built and unbuilt versions.  I reproduced it in unbuilt equality-explorer with `?brand=phet&ea&fuzz`.

Two cases of DownUpListener were replaced with FireListener in this sim, for https://github.com/phetsims/equality-explorer/issues/158.  The shas for the commits are 517290b and 7213438.  I don't see anything obviously wrong with the usages there. 

In PressListener, the failure occurs on line 671:

```js
  enter( event ) {
    sceneryLog && sceneryLog.InputListener && sceneryLog.InputListener( `PressListener#${this._id} enter` );
    sceneryLog && sceneryLog.InputListener && sceneryLog.push();

671 this.overPointers.push( event.pointer );

    sceneryLog && sceneryLog.InputListener && sceneryLog.pop();
  },
```

So based on the assertion message ("duplicates are not allowed") I'm wondering if there's a general (multitouch?) problem here with FireListener.

@jonathanolson could you please have a look?

**Comment by jonathanolson** [comment]
I believe this is probably related to https://github.com/phetsims/scenery/issues/1021, is there a point where a node with an input listener is removed? `pressListener.clearOverPointers()` is the current workaround.

**Comment by pixelzoom** [comment]
Yes, see SnapshotControl.js line 171, where `upListener` is an instance of FireListener:

```js
        this.removeInputListener( upListener );
```

This is executed when SnapshotControl has no associated snapshot, and therefore is not interactive. I can certainly use the `clearOverPointers` workaround, or work around it by not removing the listener. But it sounds like a general problem that needs to be addressed.

**Comment by pixelzoom** [comment]
I've reopened phetsims/scenery#1021 and labeled it high priority.  If we're converting to these new input listeners, then we should be able to remove a listener without things breaking, or having to resort to workarounds or reimplementing. 

In the meantime, I've reimplemented SnapshotControl to avoid removing the offending listener.  Looks OK in local fuzz testing.  I'll leave this open until CT verifies.
